Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM alpine:3.12 as builder
+
+ARG VERSION=1.0.0
+
+RUN \
+  apk add --no-cache tar curl && \
+  curl -L -O https://github.com/input-output-hk/cardano-transactions/releases/download/$VERSION/cardano-transactions-$VERSION-linux64.tar.gz && \
+  tar -C /usr -xvzf cardano-transactions-$VERSION-linux64.tar.gz bin/cardano-tx && \
+  chmod +x /usr/bin/cardano-tx && \
+  rm cardano-transactions-$VERSION-linux64.tar.gz
```

Comment:
That won't do, the goal of the Dockerfile would be to build things locally, not to download a pre-existing binary. 
Pretty much in the lights of: https://github.com/input-output-hk/cardano-addresses/blob/master/Dockerfile

Then, cabal, stack, chose your weapon. However I'd still favor stack here for consistency with cardano-addresses.

Reply:
Alright, that's possible too. I remember that Docker wizards don't really like that approach, because it's very hard to debug (similar to CI) on failures, but I guess we won't really expect many failures due to stack etc.

Reply:
Yes. Most of our users who build with Docker do it because they want to build from sources without caring about the system dependencies. But the ability to build from sources is essential. 

Arguably, Nix solves this in a nicer way, but Docker is a good answer that we ought to support too.

Reply:
Then I think we should consider caching dependencies on a public S3 bucket: https://github.com/fpco/cache-s3

Reply:
Possibly as a second iteration. Initially, just leveraging the local docker cache would be enough.

Reply:
Stack here takes about 30~ minutes to clone all source repos (and clones one repo multiple times). Cabal needs about 2 minutes max.

Reply:
After 1 hour of cloning and bulding, I get:

```
--  While building package cardano-crypto-class-2.0.0 using:
      /root/.stack/setup-exe-cache/x86_64-linux/Cabal-simple_mPHDZzAJ_2.4.0.1_ghc-8.6.5 --builddir=.stack-work/dist/x86_64-linux/Cabal-2.4.0.1 configure --user --package-db=clear --package-db=global --package-db=/root/.stack/snapshots/x86_64-linux/8ed63663e0a851b03dbe84e9d9e3c3fd980f39b6217518155664a844f84239d4/8.6.5/pkgdb --libdir=/root/.stack/snapshots/x86_64-linux/8ed63663e0a851b03dbe84e9d9e3c3fd980f39b6217518155664a844f84239d4/8.6.5/lib --bindir=/root/.stack/snapshots/x86_64-linux/8ed63663e0a851b03dbe84e9d9e3c3fd980f39b6217518155664a844f84239d4/8.6.5/bin --datadir=/root/.stack/snapshots/x86_64-linux/8ed63663e0a851b03dbe84e9d9e3c3fd980f39b6217518155664a844f84239d4/8.6.5/share --libexecdir=/root/.stack/snapshots/x86_64-linux/8ed63663e0a851b03dbe84e9d9e3c3fd980f39b6217518155664a844f84239d4/8.6.5/libexec --sysconfdir=/root/.stack/snapshots/x86_64-linux/8ed63663e0a851b03dbe84e9d9e3c3fd980f39b6217518155664a844f84239d4/8.6.5/etc --docdir=/root/.stack/snapshots/x86_64-linux/8ed63663e0a851b03dbe84e9d9e3c3fd980f39b6217518155664a844f84239d4/8.6.5/doc/cardano-crypto-class-2.0.0 --htmldir=/root/.stack/snapshots/x86_64-linux/8ed63663e0a851b03dbe84e9d9e3c3fd980f39b6217518155664a844f84239d4/8.6.5/doc/cardano-crypto-class-2.0.0 --haddockdir=/root/.stack/snapshots/x86_64-linux/8ed63663e0a851b03dbe84e9d9e3c3fd980f39b6217518155664a844f84239d4/8.6.5/doc/cardano-crypto-class-2.0.0 --dependency=aeson=aeson-1.4.6.0-EBlgS3B9Tq36MQnW8dyprT --dependency=base=base-4.12.0.0 --dependency=base16-bytestring=base16-bytestring-0.1.1.6-17atVnUhDnM13pAkKbwL6I --dependency=bytestring=bytestring-0.10.8.2 --dependency=cardano-binary=cardano-binary-1.5.0-5RfPmOHrT8m16CMfVBBr0T --dependency=cardano-prelude=cardano-prelude-0.1.0.0-GlPHUxhp48Z3krZNP7mUog --dependency=cryptonite=cryptonite-0.25-GRKCG37cD79GVqvGPiaNxJ --dependency=deepseq=deepseq-1.4.4.0 --dependency=integer-gmp=integer-gmp-1.0.2.0 --dependency=memory=memory-0.14.18-CTT9ZW74V7DJG216YjnXgb --dependency=text=text-1.2.4.0-GrUF4lk2hk02X2GpHbr2GS --dependency=transformers=transformers-0.5.6.2 --dependency=vector=vector-0.12.1.2-3ubVJHVYUZgKP2BOIpfTrf -f-development --exact-configuration --ghc-option=-fhide-source-paths
    Process exited with code: ExitFailure 1
```

This is why Dockerfile s**x for building.